packages/xforms-engine/src/client/SelectNode.ts

```diff
@@ -33,43 +52,100 @@ export interface SelectNodeState extends BaseNodeState {
 	 * Should a `SelectNodeState` have this `value` type, whereas a hypothetical
 	 * `Select1NodeState` would have `get value(): SelectItem | null`?
 	 */
-	get value(): readonly SelectItem[];
+	get value(): SelectValues<V>;
 }
 
-export interface SelectDefinition extends LeafNodeDefinition {
+export interface SelectDefinition<V extends ValueType = ValueType> extends LeafNodeDefinition<V> {
```

Comment:
I would like to think about this naming convention. Note that this currently conflicts with the definition for its `bodyElement`, which follows an earlier naming convention, and I definitely would prefer renaming that regardless of the convention for node definitions.

I'd also really like to consider removing node definitions from the engine/client API entirely. I don't think they've ever provided much value, and I continue to chip away at what little they do provide whenever I get a chance.